### PR TITLE
Retry cypress command for rate change popup - Removing Flakiness

### PR DESCRIPTION
**Pull request by usman-deriv**
### Added retry cypress command when rate is changed when confirming an order

Changes:
1. Added cypress.command inside my spec file for retry 
2. added line 'cy.c_checkRateChanged()' to call cypress command

Rest changes are **_formatting_** only on commit.

### CODE REVIEW

File: cypress/e2e/wip/P2P/blockUnblockAdvertiser.cy.js [hunk 1]
```diff
@@ -1,107 +1,201 @@
...
+            cy.get('.rate-change-modal__button').within(() => {
+              cy.get('button[type="submit"]', { log: false }).click({
+                log: false,
+              })
+            })
+            cy.findByRole('button', { name: 'Confirm' })
+              .should('exist')
+              .and('be.enabled')
+              .click()
+            cy.c_checkRateChanged({ ...options, retryCount: retryCount + 1 })
```
Comment (bilalmajid-deriv) [inline cypress/e2e/wip/P2P/blockUnblockAdvertiser.cy.js:32]: this is not required here can remove

File: cypress/e2e/wip/P2P/blockUnblockAdvertiser.cy.js [hunk 2]
```diff
@@ -1,107 +1,201 @@
...
+            options.retryCount = 1
```
Comment (bilalmajid-deriv) [inline cypress/e2e/wip/P2P/blockUnblockAdvertiser.cy.js:33]: this is also not needed here

**Review by bilalmajid-deriv — CHANGES_REQUESTED** [review]
minor changes requested

File: cypress/e2e/wip/P2P/blockUnblockAdvertiser.cy.js (hunk 2 above)
Comment (bilalmajid-deriv) [reply cypress/e2e/wip/P2P/blockUnblockAdvertiser.cy.js:33]: by this I mean 
`
cy.c_checkRateChanged({ ...options, retryCount: retryCount + 1 })
            options.retryCount = 1
`

File: cypress/e2e/wip/P2P/blockUnblockAdvertiser.cy.js [hunk 3]
```diff
@@ -1,107 +1,201 @@
...
+      if ($element.children().length > 0) {
+        cy.document({ log: false }).then((doc) => {
+          const rateChangeError = doc.querySelector(
+            '.rate-change-modal__button'
+          )
+          if (rateChangeError) {
+            cy.log(
+              `Rate change detected, waiting for ${waitTimeAfterError / 1000}s before retrying...`
+            )
+            cy.wait(waitTimeAfterError, { log: false })
```
Comment (bilalmajid-deriv) [inline cypress/e2e/wip/P2P/blockUnblockAdvertiser.cy.js:22]: this to be removed as well

File: cypress/e2e/wip/P2P/blockUnblockAdvertiser.cy.js [hunk 4]
```diff
@@ -1,107 +1,201 @@
 import '@testing-library/cypress/add-commands'
 
+Cypress.Commands.add('c_checkRateChanged', (options = {}) => {
+  const {
+    retryCount = 1,
+    maxRetries = 3,
+    waitTimeAfterError = 5000,
```
Comment (bilalmajid-deriv) [inline cypress/e2e/wip/P2P/blockUnblockAdvertiser.cy.js:7]: this as well can be removed

**Review by bilalmajid-deriv — APPROVED** [review]
LGTM

**Review by meeyun-deriv — APPROVED** [review]
LGTM

**Pull request merged**